conans/client/generators/pkg_config.py

```diff
@@ -35,7 +35,10 @@ def compiler(self):
     def content(self):
         ret = {}
         for depname, cpp_info in self.deps_build_info.dependencies:
-            name = cpp_info.name.lower() if cpp_info.name != depname else depname
+            if cpp_info.name and cpp_info.get_name("pkg_config") != cpp_info.name:
```

Comment:
Add a comment to explain. Lowercase in the general case (when name or package name) and literal otherwise.